### PR TITLE
Nav: navigation.cpp/hpp testing

### PR DESCRIPTION
**Pull request by callumTM**
## Description
Testing for `navigation.cpp/hpp`, currently being used to try and understand the system better

### CODE REVIEW

File: test/navigation/navigation.test.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  hyped::navigation::Navigation nav(log_, 0);
+
+  // unsure of how or what to set the sensors data to
+  Randomiser::randomiseSensorsData(sensors_data_);
+
+  data_.setSensorsData(sensors_data_);
+
+  // this should read the sensors data and use it to update velocity etc. but doesn't seem to
+  nav.navigate();
+  std::cout << data_.getNavigationData().velocity
```
Comment (maxguy2001) [inline test/navigation/navigation.test.cpp:33]: semicolon at the end of this line?

File: test/navigation/navigation.test.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#include "randomiser.hpp"
+#include "test.hpp"
+
+#include <iostream>
+#include <string>
+
+#include "gtest/gtest.h"
```
Comment (maxguy2001) [inline test/navigation/navigation.test.cpp:7]: '#include <gtest/gtest.h>'?

Comment (MiltFra) [reply test/navigation/navigation.test.cpp:7]: This *may* not be an issue but it is highly advised as it's part of a different binary (to some degree, let's not get into the details of this). Essentially these headers are clearly not local to the source file.

File: test/navigation/navigation.test.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  // Run before each test
+  void SetUp() {}
+  void TearDown() {}
+};
+
+namespace hyped::navigation {
+TEST_F(NavigationTest, correctEmergencyBrakingDistance)
+{
+  Data &data_ = Data::getInstance();
+  Sensors sensors_data_;
```
Comment (MiltFra) [inline test/navigation/navigation.test.cpp:23]: Why the underscore here? It's a local variable.

File: test/navigation/navigation.test.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  hyped::navigation::Navigation nav(log_, 0);
+
+  // unsure of how or what to set the sensors data to
+  Randomiser::randomiseSensorsData(sensors_data_);
```
Comment (MiltFra) [inline test/navigation/navigation.test.cpp:27]: This is fine, don't worry.

File: test/navigation/navigation.test.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  data_.setSensorsData(sensors_data_);
+
+  // this should read the sensors data and use it to update velocity etc. but doesn't seem to
```
Comment (MiltFra) [inline test/navigation/navigation.test.cpp:31]: Have you checked what the randomiser actually writes into data? Maybe the values are such that we would deterministically expect a certain velocity?

**Review by MiltFra — CHANGES_REQUESTED** [review]

File: test/navigation/navigation.test.cpp (hunk 5 above)
Comment (MiltFra) [reply test/navigation/navigation.test.cpp:31]: That's why you should put in hand-picked values because you know what to expect, by the way.

**Pull request closed without merging**